receiver/otlpreceiver/internal/arrow/arrow.go

```diff
@@ -127,41 +114,35 @@ func (r *Receiver) ArrowStream(serverStream arrowpb.ArrowStreamService_ArrowStre
 	}
 }
 
-func (r *Receiver) processRecords(ctx context.Context, records []*batchEvent.RecordMessage, producers allProducers) error {
-	for _, msg := range records {
-		switch msg.PayloadType() {
-		case arrowpb.OtlpArrowPayloadType_METRICS:
-			return ErrNoMetricsConsumer
-		case arrowpb.OtlpArrowPayloadType_LOGS:
-			return ErrNoLogsConsumer
-		case arrowpb.OtlpArrowPayloadType_SPANS:
-			if producers.Traces == nil {
-				return ErrNoTracesConsumer
-			}
-			// TODO: Use the obsreport object to instrument (somehow)
-			otlp, err := producers.Traces.ProduceFrom(msg.Record())
+func (r *Receiver) processRecords(ctx context.Context, records *arrowpb.BatchArrowRecords) error {
+	payloads := records.GetOtlpArrowPayloads()
+	if len(payloads) == 0 {
+		return nil
+	}
+	// TODO: Uncertainty about why Type is per-payload but
+	// consumer API is single-type. Should the consumer API accept
+	// arrow.Record?
+	switch payloads[0].Type {
```

Comment:
@lquerel See if this looks right (approximately).

Reply:
LGTM